shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -78,80 +98,71 @@ @interface FlutterViewController () <FlutterBinaryMessenger, UIScrollViewDelegat
+/// estimate if such an event represents the natural end of scrolling inertia or a user-initiated
+/// cancellation.
+@property(nonatomic, assign) NSTimeInterval scrollInertiaEventAppKitDeadline;
+
 /// VSyncClient for touch events delivery frame rate correction.
 ///
 /// On promotion devices(eg: iPhone13 Pro), the delivery frame rate of touch events is 60HZ
 /// but the frame rate of rendering is 120HZ, which is different and will leads jitter and laggy.
 /// With this VSyncClient, it can correct the delivery frame rate of touch events to let it keep
 /// the same with frame rate of rendering.
-@property(nonatomic, retain) VSyncClient* touchRateCorrectionVSyncClient;
+@property(nonatomic, strong) VSyncClient* touchRateCorrectionVSyncClient;
 
 /*
  * Mouse and trackpad gesture recognizers
  */
 // Mouse and trackpad hover
-@property(nonatomic, retain)
+@property(nonatomic, strong)
     UIHoverGestureRecognizer* hoverGestureRecognizer API_AVAILABLE(ios(13.4));
 // Mouse wheel scrolling
-@property(nonatomic, retain)
+@property(nonatomic, strong)
     UIPanGestureRecognizer* discreteScrollingPanGestureRecognizer API_AVAILABLE(ios(13.4));
 // Trackpad and Magic Mouse scrolling
-@property(nonatomic, retain)
+@property(nonatomic, strong)
     UIPanGestureRecognizer* continuousScrollingPanGestureRecognizer API_AVAILABLE(ios(13.4));
 // Trackpad pinching
-@property(nonatomic, retain)
+@property(nonatomic, strong)
     UIPinchGestureRecognizer* pinchGestureRecognizer API_AVAILABLE(ios(13.4));
 // Trackpad rotating
-@property(nonatomic, retain)
+@property(nonatomic, strong)
     UIRotationGestureRecognizer* rotationGestureRecognizer API_AVAILABLE(ios(13.4));
 
-/**
- * Creates and registers plugins used by this view controller.
- */
+/// Creates and registers plugins used by this view controller.
 - (void)addInternalPlugins;
 - (void)deregisterNotifications;
+
+/// Called when the first frame has been rendered. Invokes any registered first-frame callback.
+- (void)onFirstFrameRendered;
+
+/// Handles updating viewport metrics on keyboard animation.
+- (void)handleKeyboardAnimationCallbackWithTargetTime:(fml::TimePoint)targetTime;
 @end
 
 @implementation FlutterViewController {
+  // TODO(cbracken): https://github.com/flutter/flutter/issues/137801
+  // Eliminate once we can use weak pointers in platform_view_ios.h.
   std::unique_ptr<fml::WeakNSObjectFactory<FlutterViewController>> _weakFactory;
-  fml::scoped_nsobject<FlutterEngine> _engine;
-
-  // We keep a separate reference to this and create it ahead of time because we want to be able to
-  // set up a shell along with its platform view before the view has to appear.
-  fml::scoped_nsobject<FlutterView> _flutterView;
-  fml::scoped_nsobject<UIView> _splashScreenView;
-  fml::ScopedBlock<void (^)(void)> _flutterViewRenderedCallback;
-  UIInterfaceOrientationMask _orientationPreferences;
-  UIStatusBarStyle _statusBarStyle;
+  FlutterEngine* _engine;
```

Comment:
Notice that previously we held `_engine` as a strong ref (the `fml::scoped_nsobject` above) but that the header declares it as a weak property:
https://github.com/flutter/engine/blob/33f906426d067812215d46f8cad9f286ab2fdbab/shell/platform/darwin/ios/framework/Headers/FlutterViewController.h#L236

Meanwhile, the engine owns FlutterViewController weakly:
https://github.com/flutter/engine/blob/33f906426d067812215d46f8cad9f286ab2fdbab/shell/platform/darwin/ios/framework/Source/FlutterEngine.mm#L157